lib/portlayer/network/context.go

```diff
@@ -1258,6 +1258,35 @@ func (c *Context) DeleteScope(ctx context.Context, name string) error {
 		return fmt.Errorf("%s has active endpoints", s.Name())
 	}
 
+	var allzeros, allones net.IP
+	if !ip.IsUnspecifiedSubnet(s.subnet) {
+		allzeros = ip.AllZerosAddr(s.subnet)
+		allones = ip.AllOnesAddr(s.subnet)
+	}
+
+	for _, p := range s.spaces {
+		for _, i := range append(s.dns, s.gateway, allzeros, allones) {
+			if !ip.IsUnspecifiedIP(i) {
+				p.ReleaseIP4(i)
+			}
+		}
+
+		if p.Parent != nil {
+			p.Parent.ReleaseIP4Range(p)
+		}
+	}
+
+	var parentSpace *AddressSpace
+	if len(s.spaces) == 1 {
+		parentSpace = s.spaces[0]
```

Comment:
in the case where `len(s.spaces) == 1`, the only space in the slice is the parent. See this case here: https://github.com/vmware/vic/blob/master/lib/portlayer/network/context.go#L443